Create HashCommaChecker

r7kamura opened the pull request:
## About

I'd like to check the existence of the comma after the last value in a multiline Hash.

``` ruby
# Bad
{
  :a => 1,
  :b => 2,
  :c => 3
}

# Good
{
  :a => 1,
  :b => 2,
  :c => 3,
}

# Good
{ :a => 1, :b => 2, :c => 3 }
```
## Usage

Execute `/path/to/bin/guideline` or:

``` ruby
#!/usr/bin/env ruby

$LOAD_PATH.unshift File.expand_path("../../lib", __FILE__)
require "guideline"

checker = Guideline::HashCommaChecker.new
visitor = Guideline::Visitor.new(:checker => checker)
visitor.check
visitor.render
```
## Result

Here is an example result of HashCommaChecker in the rails repository.
https://gist.github.com/748a75d61e43f69f1182
## Spec

The rspec result is like following:

```
Guideline::HashCommaChecker
  #check
    when there is no last comma in sigleline hash
      checker does not detect error
    when there is last comma in sigleline hash
      checker does not detect error
    when there is no last comma in multiline hash
      checker detects error
    when there is no last comma in multiline hash with array
      checker detects error
    when there is last comma in multiline hash
      checker does not detect error
    when there is no last comma in multiline nested hash
      checker detects error

Finished in 0.00416 seconds
6 examples, 0 failures
```

Commit:
Create HashCommaChecker

sorah commented on spec/guideline/checkers/hash_comma_checker_spec.rb:
```diff
@@ -0,0 +1,102 @@
+      end
+
+      context "when there is last comma in sigleline hash" do
+        let(:content) do
```
I can't think `content` is suitable name:
- `fixture`
- `target` ( to check)
- `script`

r7kamura replied:
Thanks, I replaced it by `%s/content/script/`. I had named it in the sense of a file of content, but the word script feels better for me because the reality of the content is a ruby script. :ant:
ce77fa4

r7kamura committed:
Replace the word of content with a word script

I named it in the sense of a file of content,
but the word script feels better for me because
the reality of the content is a ruby script

r7kamura committed:
Fix wrong code added by mistake
